run_operations.py

```diff
@@ -119,168 +128,181 @@ def command_line_parse(args):
 	logger.info("\t\t--startpysar       : %s\n", inps.startpysar)
 	logger.info("\t\t--startinsarmaps   : %s\n", inps.startinsarmaps)
 	logger.info("\t\t--testsheet	    : %s\n", inps.test_sheet)
-	
 
-	
-###################### Auxiliary Functions #####################
 
+###################### Auxiliary Functions #####################
 
-"""
-    Reads template file for the current dataset and parses out the 'ssaraopt' option before creating command options line options 
-    array for ssara_federated_query.py
-    Parameters: none
-    Returns:    ssara_options: [str], array of command line options to run ssara_federated_query with
-"""
 def create_ssara_options():
-	
-	with open('/nethome/'+user+'/insarlab/OPERATIONS/TEMPLATES/'+dataset+'.template', 'r') as template_file:
+	""" Reads template file for the current dataset and parses out the 'ssaraopt' option before creating command options
+		line options array for ssara_federated_query.py
+
+		Parameters: none
+		Returns:    ssara_options: [str], array of command line options to run ssara_federated_query with
+	"""
+
+	with open('/nethome/' + user + '/insarlab/OPERATIONS/TEMPLATES/' + dataset + '.template', 'r') as template_file:
 		options = ''
 		for line in template_file:
 			if 'ssaraopt' in line:
 				options = line.strip('\n').rstrip().split("= ")[1]
 				break;
-					
+
 	# Compute SSARA options to use
 	options = options.split(' ')
 
-	ssara_options = ['ssara_federated_query.py'] + options + ['--print']	
-		
+	ssara_options = ['ssara_federated_query.py'] + options + ['--print']
+
 	return ssara_options
-		
-"""
-    Reads the most recently stored date for the given dataset from the stored_date.date file, and parses the newest data date from ssara_federated_query.
-    Parameters: ssara_output: str, string output from ssara_federated_query.py ... --print
-    Returns:    none
-"""
+
+
 def set_dates(ssara_output):
+	""" Reads the most recently stored date for the given dataset from the stored_date.date file, and parses the newest
+		data date from ssara_federated_query.
+
+		Parameters: ssara_output: str, string output from ssara_federated_query.py ... --print
+		Returns:    none
+	"""
 	global stored_date, most_recent
-	
+
 	most_recent_data = ssara_output.split("\n")[-2]
 	most_recent = datetime.strptime(most_recent_data.split(",")[3], date_format)
 
+	# Checks if the stored_date.date file exists and created the file+directories if it doesn't
+	filename = os.getenv('OPERATIONS') + '/stored_date.date'
+	if not os.path.exists(filename):
+		if not os.path.exists(os.path.dirname(filename)):
+			os.makedirs(os.path.dirname(filename))
+		subprocess.Popen("touch "+filename, shell=True).wait()
```

Comment:
Seems like a code smell to create a new file using bash. Try: https://www.tutorialspoint.com/How-to-create-an-empty-file-using-Python

Reply:
Ive tried those, and the file never gets created. Im not happy with the way that works either, but the pythonic ways aren't working.